src/main/java/io/gsonfire/util/FieldNameResolver.java

```diff
@@ -0,0 +1,69 @@
+package io.gsonfire.util;
+
+import com.google.gson.FieldNamingStrategy;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.internal.bind.ReflectiveTypeAdapterFactory;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class FieldNameResolver {
```

Comment:
There is a bug on this class. Different ```Gson``` values can be passed on ```getFieldName```, but we store only one cached ```fieldNamingStrategy```. This means that if I call ```getFieldName``` twice, with two different ```Gson``` objects, I might get the wrong strategy for the second one.

The bug could be reproducible by using a single ```GsonFireBuilder``` to create multiple ```Gson``` objects.

Since we cannot pass the ```Gson``` instance as a constructor dependency, I suggest we use a ```WeakHashMap``` to store a cache of Gson->Strategies